nav2_collision_monitor/include/nav2_collision_monitor/source_base.hpp

```diff
@@ -0,0 +1,133 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_COLLISION_MONITOR__SOURCE_BASE_HPP_
+#define NAV2_COLLISION_MONITOR__SOURCE_BASE_HPP_
+
+#include <vector>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+
+#include "tf2/time.h"
+#include "tf2_ros/buffer.h"
+
+#include "nav2_collision_monitor/types.hpp"
+
+namespace nav2_collision_monitor
+{
+
+/**
+ * @brief Basic data source class
+ */
+class SourceBase
+{
+public:
+  /**
+   * @brief SourceBase constructor
+   */
+  SourceBase(
+    const nav2_util::LifecycleNode::WeakPtr & node,
+    std::shared_ptr<tf2_ros::Buffer> tf_buffer,
+    const std::string & source_name,
+    const std::string & base_frame_id,
+    const tf2::Duration & transform_tolerance,
+    const tf2::Duration & data_timeout);
+  /**
+   * @brief SourceBase destructor
+   */
+  virtual ~SourceBase();
+
+  /**
+   * @brief Data source configuration routine.
+   * Empty virtual method intended to be used in child implementations.
+   */
+  virtual void configure() = 0;
+
+  /**
+   * @brief Sets fixed frame ID. Used in data frame interpolation
+   * @param fixed_frame_id Fixed frame ID
+   */
+  void setFixedFrameId(const std::string & fixed_frame_id);
+
+  /**
+   * @brief Adds latest data from source to the data array.
+   * Empty virtual method intended to be used in child implementations.
+   * @param curr_time Current node time for data interpolation
+   * @param data Array where the data from source to be added.
+   * Added data is converted to base_frame_id coordinate system at curr_time.
+   */
+  virtual void getData(
+    const rclcpp::Time & curr_time, std::vector<Point> & data) = 0;
+
+protected:
+  /**
+   * @brief Supporting routine obtaining all ROS-parameters
+   * @param source_topic Output name of source subscription topic
+   */
+  void getParameters(std::string & source_topic);
+
+  /**
+   * @brief Checks whether the source might be considered as valid
+   * @param curr_time Current node time for source verification
+   * @return True if data source is valid, otherwise false
+   */
+  bool sourceValid(const rclcpp::Time & curr_time);
+
+  /**
+   * @brief Obtains a transform from source_frame_id at source_time ->
+   * to base_frame_id at current node time
+   * @param curr_time Current node timestamp to interpolate data to
+   * @param source_time Source timestamp to convert data from
+   * @param tf2_transform Output source->base transform
+   * @return True if got correct transform, otherwise false
+   */
+  bool getSourceBaseTransform(
```

Comment:
`getTransform()`? `getTransformToGlobalFrame()`? The current naming is a bit confusing

Reply:
[Renamed](https://github.com/ros-planning/navigation2/pull/2982/files#diff-80488474a2ed64b56c64dcb02ce35b76423899f21798f978e5f64463f8a06729R73) to `getTransform()`